salt/engines/__init__.py

```diff
@@ -46,10 +46,22 @@ def start_engines(opts, proc_mgr, proxy=None):
             engine, engine_opts = next(iter(engine.items()))
         else:
             engine_opts = None
-        fun = '{0}.start'.format(engine)
+        engine_name = None
+        if engine_opts is not None and 'type' in engine_opts:
```

Comment:
where is that 'type' coming from? It's a new configuration for an engine?

Reply:
`type` would be a new configuration option in the engine configuration, so it would contain strings like `slack` or `script` depending on what type of engine the particular stanza in the configuration would be providing.